flag_test.go

```diff
@@ -658,6 +658,37 @@ func TestNormalizationFuncShouldChangeFlagName(t *testing.T) {
 	}
 }
 
+// This is a situation encountered when using the cobra package
+func TestNormalizationSharedFlags(t *testing.T) {
+	f := NewFlagSet("set f", ContinueOnError)
+	g := NewFlagSet("set g", ContinueOnError)
+
+	f.Bool("valid_flag", false, "bool value")
+	g.AddFlagSet(f)
+
+	f.SetNormalizeFunc(wordSepNormalizeFunc)
+	g.SetNormalizeFunc(wordSepNormalizeFunc)
+
```

Comment:
Please add a check if flag names are really normalized:
```go
if f.Lookup("valid.flag") == nil {
    t.Error("Name of valid_flag is not normalized")
}
```

Reply:
All right. If this is not working, the previous test should pick it up, but I suppose it could make a failure of this test less clear. Maybe can/should even check explicitly the value of the key in the underlying map.

Reply:
`Maybe can/should even check explicitly the value of the key in the underlying map.`
Yes, it will be even better.